src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/KinesisDataFetcher.java

```diff
@@ -169,6 +193,11 @@ void advanceIteratorTo(String sequenceNumber, InitialPositionInStreamExtended in
             nextIterator = getIterator(ShardIteratorType.AT_SEQUENCE_NUMBER.toString(), sequenceNumber);
         }
         if (nextIterator == null) {
+            final Set<String> childShardIds = leaseCoordinator.getChildShardIds(shardId);
+            if (CollectionUtils.isNullOrEmpty(childShardIds)) {
+                throw new InvalidStateException("Failed to advance iterator for shard " + shardId
```

Comment:
We persist shard end for ResourceNotFoundException as well right? How are we differentiating this? any implications on just logging this?

Reply:
ResourceNotFoundException should happen when the stream is no longer existing. So for this case, childShards propagated to ShutdownTask would be an empty list. So the childShards in DDB table would also be empty list.

Reply:
ResourceNotFound can happen either when stream is not found or when shard is trimmed. But ya in either of these cases, we can persist just the shard end.